File: lab2-functionalPrimer/javascript/func_as_arg.js
function arrFunc(a, b) {
  let arr = [];
  for (let i = a; i <= b; i++) {
    arr.push(i);
  }
  return arr;
}

function applicatorFunc(inpFunc, s, a, b) {
  const arr = inpFunc(a, b);
  const sum = arr.reduce((acc, val) => acc + val, 0);

  if (s === 's') {
    return sum;
  } else {
    return sum / (b - a + 1);
  }
}

console.log(applicatorFunc(arrFunc, 's', 1, 10));

File: lab2-functionalPrimer/python3/func_as_arg.py
def listFunc(a, b):
    return [i for i in range(a, b + 1)]

def applicatorFunc(inpFunc, s, a, b):
    if s == 's':
        return sum(inpFunc(a, b))
    else:
        return sum(inpFunc(a, b)) / (b - a + 1)

print(applicatorFunc(listFunc, 's', 1, 10))
